akka-http-core/src/main/scala/akka/http/impl/engine/ws/Handshake.scala

```diff
@@ -57,69 +64,202 @@ private[http] object Handshake {
             list of values indicating which extensions the client would like
             to speak.  The interpretation of this header field is discussed
             in Section 9.1.
-   */
-  def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
-    def find[T <: HttpHeader: ClassTag]: Option[T] =
-      headers.collectFirst {
-        case t: T ⇒ t
+    */
+    def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
+      def find[T <: HttpHeader: ClassTag]: Option[T] =
+        headers.collectFirst {
+          case t: T ⇒ t
+        }
+
+      // val host = find[Host]
```

Comment:
Basically, I mention all the things here from the list of the headers from the RFC mentioned above. I should maybe add  more comments why things don't need to be checked at this place because they will be validated at another place.